### PR TITLE
Ospp 2024/feat schema render

### PR DESCRIPTION
**Pull request by Fleurxxx**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Release Notes

- **New Features**
	- Comprehensive user guide for interacting with the AI chat interface, including configuration of AI access tokens.
	- Enhanced chat functionality with file upload support for images and speech recognition capabilities.
	- Real-time code generation from image uploads and descriptions.

- **Improvements**
	- Updated button styling for better user experience.
	- Enhanced state management for AI model interactions and token handling.

- **Bug Fixes**
	- Improved error handling for file uploads and token validation.

### CODE REVIEW

**Comment by CatsAndMice** [comment]
@Fleurxxx 描述要认真写
image

File: packages/plugins/robot/src/Main.vue [hunk 1]
```diff
@@ -189,15 +189,82 @@ export default {
...
-     每次回复请遵循以下准则：
-        1. 如果需要展示代码，确保回复中只包含一个代码块。
-        2. 所有代码必须基于 Vue 3 框架编写。
-        3. 所有使用的组件必须来自 TinyVue 组件库，严禁使用 Element UI 等其他第三方组件库或原生组件。例如，想使用输入框组件，应该使用TinyVue组件库中的 \`tiny-input\`;想使用按钮组件，应该使用 TinyVue 组件库中的 \`tiny-button\`。
-        4. 仔细阅读并遵循 [TinyVue 组件库文档](https://opentiny.design/tiny-vue/zh-CN/os-theme/overview) 中的指导，确保代码的准确性和一致性。
-     请根据上述准则，使用 TinyVue 组件库生成高质量的前端代码。
-     `
+    const codeRules = `请生成一个JSON格式的schema代码，确保它完全符合以下提供的模板。请严格按照模板的结构和属性来构建代码，不要添加任何额外的属性或元素，也不要省略任何必要的部分。以下是您需要遵循的模板：
+
+{
```
Comment (hexqi) [inline packages/plugins/robot/src/Main.vue]: 最好是先有一段描述，简短介绍下格式，然后才是下面的这个示例，这样效果可能会好点

File: packages/plugins/robot/src/Main.vue [hunk 2]
```diff
@@ -189,15 +189,82 @@ export default {
...
+  "state": {
+    "dataDisk": [1, 2, 3]
+  },
+  "methods": {},
+  "componentName": "Page",
+  "css": "body { background-color: #eef0f5; margin-bottom: 80px; }",
+  "props": {},
```
Comment (hexqi) [inline packages/plugins/robot/src/Main.vue:201]: 有一些没有必要的信息可以不用在示例里面带上，比如fileName、props

File: packages/plugins/robot/src/Main.vue [hunk 3]
```diff
@@ -189,15 +189,82 @@ export default {
...
-        1. 如果需要展示代码，确保回复中只包含一个代码块。
-        2. 所有代码必须基于 Vue 3 框架编写。
-        3. 所有使用的组件必须来自 TinyVue 组件库，严禁使用 Element UI 等其他第三方组件库或原生组件。例如，想使用输入框组件，应该使用TinyVue组件库中的 \`tiny-input\`;想使用按钮组件，应该使用 TinyVue 组件库中的 \`tiny-button\`。
-        4. 仔细阅读并遵循 [TinyVue 组件库文档](https://opentiny.design/tiny-vue/zh-CN/os-theme/overview) 中的指导，确保代码的准确性和一致性。
-     请根据上述准则，使用 TinyVue 组件库生成高质量的前端代码。
-     `
+    const codeRules = `请生成一个JSON格式的schema代码，确保它完全符合以下提供的模板。请严格按照模板的结构和属性来构建代码，不要添加任何额外的属性或元素，也不要省略任何必要的部分。以下是您需要遵循的模板：
+
+{
+  "state": {
```
Comment (hexqi) [inline packages/plugins/robot/src/Main.vue]: 有些字段是生成代码中很关键的，像children，还有几个相对关键的，比如css,state,methods，这些在Prompt中可以介绍下作用和格式；另外有些不需要AI关注的字段其实在示例中可以去掉，比如fileName,props等，后面AI返回结果可以再跟一个默认值做合并

File: packages/plugins/robot/src/Main.vue [hunk 4]
```diff
@@ -189,15 +189,82 @@ export default {
...
+      ]
+    },
+    {
+      "componentName": "div",
+      "props": {
+        "style": "border-width: 1px; border-style: solid; border-radius: 4px; border-color: #fff; padding: 10px; box-shadow: rgba(0, 0, 0, 0.1) 0px 1px 3px 0px; background-color: #fff; margin-bottom: 10px;"
+      },
+      "children": [
+        {
+          "componentName": "TinyForm",
```
Comment (hexqi) [inline packages/plugins/robot/src/Main.vue]: 关于组件这块的信息也是在Prompt中需要关注的

**Review by CatsAndMice — APPROVED** [review]

**Comment by Fleurxxx** [comment]
@chilingling 老师可以帮我合入一下这个pr吗

**Comment by Fleurxxx** [comment]
@hexqi 老师可以帮我合入一下这个pr吗

**Review by chilingling — DISMISSED** [review]

**Review by hexqi — APPROVED** [review]

**Review by hexqi — CHANGES_REQUESTED** [review]
部分commit没有签名

**Comment by Fleurxxx** [comment]
@hexqi 老师在现在全部签名了

**Review by hexqi — DISMISSED** [review]

**Review by hexqi — APPROVED** [review]

**Pull request merged**